docs/source/algorithms/SortXAxis-v2.rst

```diff
@@ -0,0 +1,76 @@
+.. algorithm::
+
+.. summary::
+
+.. relatedalgorithms::
+
+.. properties::
+
+Description
+-----------
+
+Clones the input :ref:`Matrix Workspaces <MatrixWorkspace>` and orders the
+x-axis in an ascending or descending fashion. Ensures that the y-axis and error data as well as optional Dx data
+are sorted in a consistent way with the x-axis.
+
+This algorithm is for use with workspaces of all sizes loaded into mantid (with varying speeds). It is
```

Comment:
This is only my opinion and you may disagree (and forget about this): I think sorting does not need such explicit usage examples. In the worst case one needs to read the test. If you say: for use with workspaces of all sizes this implies that other algorithms cannot be used with workspaces of all sizes. That sounds poor for Mantid users. If you say something about speed, this needs to be quantified in my opinion (which machine, how many cores used, when does it vary, it varies??). There is the word `loaded` in this sentence, I think `existing` or `present` match better.

Reply:
The reasoning for the wording there, is because if you check the SortXAxis-v1.rst which is the one wrote for the Python algorithm, you will find that it says it is for use with small workspaces, I have found through usage of both algorithms that it works just fine at least in C++ form faster on workspaces of the same size and I am fairly sure that it can run on larger workspaces in a "User-friendly time" which is to say it won't take many hours. I am not saying the python one did but the C++ from my limited exposure actually is a faster running algorithm, I am going to set up some stuff to benchmark my version of SortX the other C++ version and the python version, this should tell me which is faster and such, allowing a probably more accurate description. 

I do however understand why you dislike the wording and I'll try to re-word it better, because I agree it doesn't make as much sense and I was pulled up on that by the first reviewer as well on it, just in person.